src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -972,8 +965,8 @@ public function getDates()
         ];
 
         return $this->usesTimestamps()
-                    ? array_unique(array_merge($this->dates, $defaults))
-                    : $this->dates;
+                    ? $defaults
+                    : [];
```

Comment:
Also we could cast the created_at and updated_at columns as well and remove all related code from this class like the getDates().

Probaly the `isDateAttribute` method could be removed to and replaced with `isDateCastable`.